R/aggregate_kicking_stats.R

```diff
@@ -0,0 +1,161 @@
+      .groups = "drop"
+    )
+
+
+  # Extra points
+  df_pat <- df_fg_or_pat %>%
+    dplyr::filter(extra_point_attempt == 1) %>%
+    dplyr::group_by(!!!grp_vars) %>%
+    dplyr::summarise(
+      pat_made = sum(pat_res == "good", na.rm = TRUE),
+      pat_att = sum(extra_point_attempt, na.rm = TRUE),
+      pat_missed = sum(pat_res == "failed", na.rm = TRUE),
+      pat_blocked = sum(pat_res == "blocked", na.rm = TRUE),
+      pat_pct = round(pat_made / pat_att, 3L),
+      .groups = "drop"
+    )
+
+
+  # The Game Winning kicks distance include up to one value at the weekly level
+  # but can include multiple across the season. This is one way to account for that.
+  # the downside is that the column names change depending on if it is weekly vs
+  # seasonal.
+  if(weekly) {
+    gw_dist_name <- "gwfg_distance"
+  } else {
+    gw_dist_name <- "gwfg_distance_list"
+  }
+
+  # See the above note. I wonder if this shoudl also include field goals that tie
+  # the game but I kept the filter dplyr::between(score_differential, -2, 0) the way
+  # that is was previously. If you do include field goals that send the game into OT,
+  # then you'll probably need to include the gwfg_distance AND gwfg_distance_list columns
+  # in the weekly data
+  # Also, unusual one for me: An in-line if() statement.
+  game_winners <- df_fg_or_pat %>%
+    dplyr::group_by(game_id, team) %>%
+    dplyr::filter(fixed_drive == max(fixed_drive, na.rm = TRUE)) %>%
+    dplyr::ungroup() %>%
+    dplyr::filter(
+      field_goal_attempt == 1,
+      dplyr::between(score_differential, -2, 0)
+    ) %>%
+    dplyr::group_by(!!!grp_vars) %>%
+    dplyr::summarise(
+      gwfg_att = n(),
+      !!gw_dist_name := if(weekly) dist else paste(na.omit(dist), collapse = ";"),
+      gwfg_made = sum(fg_res == "made", na.rm = TRUE),
+      gwfg_missed = sum(fg_res == "missed", na.rm = TRUE),
+      gwfg_blocked = sum(fg_res == "blocked", na.rm = TRUE),
+      .groups = "drop"
+    )
+
+  # Prepping data to merge-in player names. Not 100% sure how you wanted this
+  # done so feel free to scrap the way I did this
+  df_player_names <- df_fg_or_pat %>%
+    dplyr::select(!!!grp_vars, player_name) %>%
+    dplyr::distinct()
+
+  # Joining all the data together and grabbing the first few columns.
+  # I was not sure what order you want the columns or if you want them arranged (and how)
+  full_kicks <- df_field_goals %>%
+    dplyr::full_join(df_pat, as.character(grp_vars)) %>%
+    dplyr::full_join(game_winners, as.character(grp_vars)) %>%
+    dplyr::left_join(df_player_names, as.character(grp_vars)) %>%
+    dplyr::select(season, any_of("week"), season_type, team, player_name, player_id, everything())
+
+  # This just replaced NAs with 0. I personally prefer the NA's but saw you explicity change
+  # NA to zero in the PR. Also, I changed character values with zero length to 0 as well
+  full_kicks[] <- lapply(full_kicks, function(x) {
```

Comment:
I could probably convinced that NAs should stay NAs.

Reply:
Ok, I changed them all to NA with the exception of the attempts columns (which are zero for games they have zero attempts for the respective kick type)